src/test/java/harness/TestUser.java

```diff
@@ -62,27 +78,92 @@
+    writeTerraOAuthCredentialFile(googleCredentials);
+
+    // We're not using pet SA key file (for security reasons), so auth is more complicated.
+    writeBqCredentialFile();
+    writeGsUtilCredentialFile();
+
+    // unset the current user in the global context if already specified
+    Context.setUser(null);
 
+    // do the login flow to populate the global context with the current user
+    User.login();
+  }
+
+  /** Writes .terra/StoredCredential. */
+  private void writeTerraOAuthCredentialFile(GoogleCredentials googleCredentials)
+      throws IOException {
     // use the domain-wide delegated credential to build a stored credential for the test user
     StoredCredential dwdStoredCredential = new StoredCredential();
-    dwdStoredCredential.setAccessToken(delegatedUserCredential.getAccessToken().getTokenValue());
+    dwdStoredCredential.setAccessToken(googleCredentials.getAccessToken().getTokenValue());
     dwdStoredCredential.setExpirationTimeMilliseconds(
-        delegatedUserCredential.getAccessToken().getExpirationTime().getTime());
+        googleCredentials.getAccessToken().getExpirationTime().getTime());
 
     // update the credential store on disk
     // set the single entry to the stored credential for the test user
     DataStore<StoredCredential> dataStore = getCredentialStore();
     dataStore.set(GoogleOauth.CREDENTIAL_STORE_KEY, dwdStoredCredential);
+  }
 
-    // unset the current user in the global context if already specified
-    Context.setUser(null);
+  /**
+   * Writes ~/.config/gcloud/legacy_credentials/default/adc.json that
+   * cloudsdk/component_build/wrapper_scripts/bq.py expects.
+   *
+   * <p>Two things are needed for `bq` auth:
+   *
+   * <ol>
+   *   <li>CLOUDSDK_AUTH_ACCESS_TOKEN (set in DockerCommandRunner.java)
+   *   <li>adc.json
+   * </ol>
+   *
+   * <p>In theory, only CLOUDSDK_AUTH_ACCESS_TOKEN should be needed. However, when support for
+   * CLOUDSDK_AUTH_ACCESS_TOKEN was added
+   * (https://cloud.google.com/sdk/docs/release-notes#cloud_sdk_2), it did not include `bq.py`. So
+   * `bq.py` still expects `adc.json`.
+   */
+  private void writeBqCredentialFile() throws IOException {
+    JsonObject json = new JsonObject();
+    json.addProperty("client_id", GCLOUD_CLIENT_ID);
+    json.addProperty("client_secret", GCLOUD_CLIENT_SECRET);
+    json.addProperty("refresh_token", getRefreshToken());
+    json.addProperty("type", "authorized_user");
+    Path path =
+        Path.of(
+            System.getProperty("user.home"), ".config/gcloud/legacy_credentials/default/adc.json");
+    Files.createDirectories(path.getParent());
+    Files.write(path, json.toString().getBytes(), StandardOpenOption.CREATE);
+  }
 
-    // do the login flow to populate the global context with the current user
-    User.login();
+  /**
+   * Writes ~/.config/gcloud/legacy_credentials/default/.boto that
+   * google-cloud-sdk/bin/bootstrapping/gsutil.py expects.
+   *
```

Comment:
I don't understand this comment.  It looks like you're saying .boto is needed both above and below?

Reply:
Deleted below